src/Sobol.jl

```diff
@@ -1,67 +1,93 @@
 module Sobol
 
-import NLopt.libnlopt
-import Base: ndims, length, skip, next, start, done, show, writemime
+import Base: ndims, skip, next, start, done, show, writemime
 export SobolSeq, ScaledSobolSeq, next!
 
-type SobolSeq{N}
-    p::Ptr{Void} # nlopt_sobol pointer
-    function SobolSeq()
-        (N < 1 || N > 1111) && error("invalid Sobol dimension")
-        p = ccall((:nlopt_sobol_create,libnlopt), Ptr{Void}, (Cuint,), N)
-        p == C_NULL && error("error initializing Sobol sequence")
-        s = new(p)
-        finalizer(s, destroy)
-        s
-    end
+include("soboldata.jl") #loads `sobol_a` and `sobol_minit`
+
+type SobolSeq
+    sdim::Int #dimension of sequence being generated
+    m::Array{UInt,2} #array of size (sdim, WORD_SIZE)
+    x::Array{UInt,1} #previous x = x_n, array of length sdim
+    b::Array{UInt,1} #position of fixed point in x[i] is after bit b[i]
+    n::UInt #number of x's generated so far
 end
-SobolSeq(N::Integer) = SobolSeq{N}()
 
-destroy(s::SobolSeq) = 
-    ccall((:nlopt_sobol_destroy,libnlopt), Void, (Ptr{Void},), s.p)
+ndims(s::SobolSeq) = s.sdim
 
-ndims{N}(s::SobolSeq{N}) = N
+function SobolSeq(N::Integer)
+    (N < 1 || N > 1111) && error("invalid Sobol dimension")
 
-function show{N}(io::IO, s::SobolSeq{N})
-    print(io, "$N-dimensional Sobol sequence on [0,1]^$N")
-end
-function writemime{N}(io::IO, ::MIME"text/html", s::SobolSeq{N})
-    print(io, "$N-dimensional Sobol sequence on [0,1]<sup>$N</sup>")
-end
+    m = ones(UInt, (N, WORD_SIZE))
 
-function skip!{N}(s::SobolSeq{N}, n::Integer, x::Vector{Float64})
-    (n < 0 || length(x) < N) && throw(BoundsError())
-    ccall((:nlopt_sobol_skip,libnlopt), Void, (Ptr{Void}, Cuint, Ptr{Float64}),
-          s.p, n, x)
-    x
-end
+    if N == 1 #special case
+        return SobolSeq(N,m,UInt[0],UInt[0],zero(UInt))
+    end
 
-function skip{N}(s::SobolSeq{N}, n::Integer) 
-    skip!(s, n, Array(Float64, N))
-    nothing
+    for i = 2:N
+        a = sobol_a[i-1]
+        d = floor(UInt, log2(a)) #degree of poly
+
+        #set initial values of m from table
+        # TODO transpose sobol_minit for faster column access?
+        m[i, 1:Int(d)] = sobol_minit[i-1, 1:Int(d)]
+        #fill in remaining values using recurrence
+        for j = (d+1):WORD_SIZE
+            ac = a
+            m[i,j] = m[i,j-d]
+            for k = 0:d-1
+                m[i,j] $= ((ac & one(UInt)) * m[i, j-d+k]) << (d-k)
+                ac >>= 1
+            end
+        end
+    end
+    SobolSeq(N,m,zeros(UInt,N),zeros(UInt,N),zero(UInt))
 end
 
-function next!{N}(s::SobolSeq{N}, x::Vector{Float64})
-    length(x) < N && throw(BoundsError())
-    ccall((:nlopt_sobol_next01,libnlopt), Void,
-          (Ptr{Void}, Ptr{Float64}), s.p, x)
-    x
+function next!(s::SobolSeq, x::Vector{Float64})
```

Comment:
Should be `AbstractVector`, not `Vector` — no reason not to support other types.  And probably `AbstractVector{T}` where `T<:Union(Float16,Float32,Float64)`, since we might as well support storing in lower precision (the `x[i] = ...` assignment will discard the extra bits).